mne_bids/write.py

```diff
@@ -2568,7 +2598,8 @@ def anonymize_dataset(bids_root_in, bids_root_out, daysback='auto',
                 bids_path=bp_out,
                 anonymize={
                     'daysback': daysback,
-                    'keep_his': False
+                    'keep_his': False,
+                    'keep_src': False,
```

Comment:
The dict keeps getting larger :) I'd propose to switch to a TypedDict now (with `total=False`)

Reply:
woooooo TIL

Reply:
About `total` or what do you mean? :)

Reply:
TypedDict in general! and `total` is a cool feature ^_^

Reply:
Hmm... how would we implement this? 

We want users to pass in a TypedDict?, or we convert `anonymize` to a TypedDict of type `class Anonymize(TypedDict, total=False)`?

Reply:
Yeah not really sure how you meant to use TypedDict here. Any suggestions @hoechenberger ?

Reply:
@adam2392, I think that could be done in a separate, dedicated PR, unless @hoechenberger disagrees.

Reply:
Wait I can't do TypedDict in Python3.7 anyways: https://www.python.org/dev/peps/pep-0589/

I'll remove this, and we can table this PR until Python 3.8 is minimal version?

Reply:
Now that 3.10 is out, we can actually start considering to ditch 3.7 … Otherwise we'll have to support 4 versions?? 🤔

Reply:
I'm down for it. I only use environment in 3.8 nowadays.

Perhaps we can discuss in the other issue?